### PR TITLE
comparison report api calls in threads

### PR DESCRIPTION
**Pull request by petrblaho**
- Refactors getting of data to submethods
- Uses intermediate dict to collect api results
- Passes current app object to get_* methods
- Moves timing code into get_* methods
- Uses ThreadPoolExecutor to run get_* methods

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by fellipeh — APPROVED** [review]

File: drift/views/v1.py [hunk 1]
```diff
@@ -194,47 +196,71 @@ def comparison_report(
...
-            except RBACDenied as error:
-                message = error.message
-                current_app.logger.audit(str(HTTPStatus.FORBIDDEN) + " " + message, request=request)
-                raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                futures = []
+                app = current_app._get_current_object()
+                if system_ids:
+                    # can raise RBACDenied exception
+                    futures.append(executor.submit(get_systems_with_profiles, app, PT_CR_API_SYSTEM_PROFILE_REQUESTS, system_ids, auth_key, current_app.logger, get_event_counters()))
```
Comment (vkrizan) [inline drift/views/v1.py:245]: You can swap the array with dict and use keys as the Future and a value noting which api result it represents.

File: drift/views/v1.py [hunk 2]
```diff
@@ -235,34 +237,25 @@ def get_historical_sys_profiles(app, timer, historical_sys_profile_ids, auth_key
...
-                        auth_key,
-                        current_app.logger,
-                        get_event_counters(),
-                    ))
-            except RBACDenied as error:
-                message = error.message
-                current_app.logger.audit(str(HTTPStatus.FORBIDDEN) + " " + message, request=request)
-                raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+                    futures.append(executor.submit(get_historical_sys_profiles, app, PT_CR_API_HSP_REQUESTS, historical_sys_profile_ids, auth_key, current_app.logger, get_event_counters()))
+                for future in concurrent.futures.as_completed(futures):
```
Comment (vkrizan) [inline drift/views/v1.py:252]: Can this be outside of the executor context?

**Comment by petrblaho** [comment]
DRFT-1179

**Comment by petrblaho** [comment]
Squashed into 2 commit

**Review by vkrizan — APPROVED** [review]

**Pull request merged**